src/pudl/analysis/plant_parts_eia.py

```diff
@@ -787,20 +826,18 @@ def add_additonal_cols(self, plant_parts_eia):
         return plant_parts_eia
 
     def _clean_plant_parts(self, plant_parts_eia):
-        plant_parts_eia = (
-            plant_parts_eia.assign(
-                report_year=lambda x: x.report_date.dt.year,
-                plant_id_report_year=lambda x: x.plant_id_pudl.astype(str)
-                + "_"
-                + x.report_year.astype(str),
-            )
-            .pipe(
-                pudl.helpers.cleanstrings_snake,
-                ["record_id_eia", "appro_record_id_eia"],
-            )
-            .set_index("record_id_eia")
```

Comment:
You can't write a table to the DB where the index is a non-numeric column, right? I ended up just making `record_id_eia` a column and then setting it to be the index in the FERC to EIA match

Reply:
I could be wrong but I think the index column is ignored entirely when writing to the DB.  The schema for the table which is defined in the metadata needs to indicate what set of columns comprise the primary key if there is one.  If there's no natural primary key then I think it'll it'll use an autoincrementing integer key, which is what happens with the `fuel_receipts_costs` table(s).